models/git/commit_status.go

```diff
@@ -254,17 +254,17 @@ func listCommitStatusesStatement(ctx context.Context, repo *repo_model.Repositor
 func sortCommitStatusesSession(sess *xorm.Session, sortType string) {
 	switch sortType {
 	case "oldest":
-		sess.Asc("created_unix")
+		sess.OrderBy(db.SearchOrderByOldest.String())
```

Comment:
How about rewrite this function like the previous `GetSearchOrderByBySortType` implementation?

Reply:
I think the sorting rules are independent for each model/table, the way I wrote it I just want to reuse some common simple sorting rules. If we use `GetSearchOrderByBySortType`, then each model needs to write a `GetSearchOrderByBySortType`, and then `db.SearchOrderBy` needs to contain all the sorting rules, which is overdesigned.
While sortType is of type `string`, we can sortType to choose to reuse `db.SearchOrderBy` or define our own sorting rules. For example:

```
switch sortType {
	case "oldest":
		sess.OrderBy(db.SearchOrderByOldest.String())
       case "nearduedate":
               sess.Join("LEFT", "milestone", "issue.milestone_id = milestone.id").
			OrderBy("CASE " +
				"WHEN issue.deadline_unix = 0 AND (milestone.deadline_unix = 0 OR milestone.deadline_unix IS NULL) THEN 253370764800 " +
				"WHEN milestone.deadline_unix = 0 OR milestone.deadline_unix IS NULL THEN issue.deadline_unix " +
				"WHEN milestone.deadline_unix < issue.deadline_unix OR issue.deadline_unix = 0 THEN milestone.deadline_unix " +
				"ELSE issue.deadline_unix END ASC").
			Desc("issue.created_unix").
			Desc("issue.id")
```